frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -227,6 +233,24 @@ function EmbeddingDataPicker({
   const context = useNotebookContext();
   const modelList = getModelFilterList(context, hasMetrics);
 
+  const metadata = useSelector(getMetadata);
+  const databases = useMemo(() => {
+    // We're joining data
+    if (!canChangeDatabase) {
+      return [
+        metadata.database(databaseId),
+        metadata.savedQuestionsDatabase(),
+      ].filter(Boolean);
```

Comment:
I initially followed Alex P.' suggestion by returning only 1 DB, but that [failed other tests in `interactive-embedding`](https://github.com/metabase/metabase/actions/runs/13560478311/job/37905551273?pr=54272), so now I use the same code we have before.
https://github.com/metabase/metabase/blob/d9507f2af8741a8389c78324b5871a8a6266c2a2/frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/JoinTablePicker/JoinTablePicker.tsx#L46-L53

I re-ran those failed tests locally, and they all passed with this version.